### PR TITLE
(DOCS-15100): Add use cases for why query MongoDB remotely

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCS-15100

### Staged Changes (Requires MongoDB Corp SSO)

- [Query MongoDB - Node.js SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCS-15100/sdk/node/examples/query-mongodb/#use-cases)
- [Query MongoDB - RN SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCS-15100/sdk/react-native/examples/query-mongodb/)
- [Query MongoDB - Java SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCS-15100/sdk/java/examples/mongodb-remote-access/#use-cases)
- [Query MongoDB - Swift SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCS-15100/sdk/swift/examples/mongodb-remote-access/#use-cases)
- [Query MongoDB - .Net SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCS-15100/sdk/dotnet/examples/mongodb-remote-access/#use-cases)

Plus a couple of ' fixes on the custom user data pages.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/dotnet/examples/mongodb-remote-access.txt [hunk 1]
```diff
@@ -15,6 +15,24 @@ from a .NET application using the {+service-short+} .NET SDK.
...
+
+There are a variety of reasons you might want to query a MongoDB data source. 
+Working with data in your client via Realm Sync is not always practical or 
+possible. You might want to query MongoDB when:
+
+- The data set is large or the client device has constraints against loading 
+  the entire data set
+- You are :ref:`creating or updating custom user data 
+  <dotnet-custom-user-data>`
+- You are retrieving results that are not modeled in Realm
```
Comment (mongodben) [inline source/sdk/dotnet/examples/mongodb-remote-access.txt:29]:
```suggestion
- You are retrieving documents that are not modeled in Realm
```

ditto for all

File: source/sdk/dotnet/examples/mongodb-remote-access.txt [hunk 2]
```diff
@@ -15,6 +15,24 @@ from a .NET application using the {+service-short+} .NET SDK.
...
+- Your app needs to access collections that don't have strict schemas
+- A non-Realm service generates collections regularly that you want to access
```
Comment (mongodben) [inline source/sdk/dotnet/examples/mongodb-remote-access.txt:31]:
```suggestion
- A non-Realm service generates collections that you want to access
```

i don't think we need 'regularly' here. don't think frequency matters. ditto for all sdks

File: source/sdk/java/advanced-guides/custom-user-data.txt [hunk 3]
```diff
@@ -23,7 +23,7 @@ User Data <custom-user-data>`.
 .. _java-read-a-users-custom-data:
 .. _java-access-custom-user-data:
 
-Read a User’s Custom Data
+Read a User's Custom Data
 -------------------------
```
Comment (mongodben) [inline source/sdk/java/advanced-guides/custom-user-data.txt:27]: nice spot of the ver subtle typo. maybe should be updated to 'Read Custom User Data'?  sounds more natural to me. same as for all the other instances of '...a User's Custom Data'.

**Review by mongodben — APPROVED** [review]
lgtm w some small comments

File: source/sdk/java/advanced-guides/custom-user-data.txt (hunk 3 above)
Comment (dacharyc) [reply source/sdk/java/advanced-guides/custom-user-data.txt:27]: I wish I could take credit, but VSCode has started highlighting these and I have no idea why. I've just been fixing them when I spot them. I think this is worded this way as a contrast to User Metadata - i.e. "User Metadata" vs. "User Custom Data." Custom Data is something we can create/edit, Metadata is something that comes through auth providers. Not sure I want to tackle a larger renaming project on that now. Interesting idea, though!

**Pull request merged**